com/course/zenkouv/src/Lecture4/task2/factory/WriterFactory.java

```diff
@@ -0,0 +1,48 @@
+package Lecture4.task2.factory;
+
+import Lecture3.Task1.model.Armor;
+import Lecture3.Task1.model.Weapon;
+import Lecture4.task2.util.Path;
+import Lecture4.task2.util.writer.ArmorWriter;
+import Lecture4.task2.util.writer.CustomWriter;
+import Lecture4.task2.util.writer.MessageWriter;
+import Lecture4.task2.util.writer.WeaponWriter;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+public class WriterFactory {
+
+    private List<CustomWriter> writers;
+
+    public WriterFactory() throws IOException {
+        try {
+            writers = Arrays.asList(new MessageWriter(Path.MESSAGE_PATH), new WeaponWriter(Path.WEAPON_PATH), new ArmorWriter(Path.ARMOR_PATH));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public List<CustomWriter> getWriters() {
+        return writers;
+    }
+
+    public CustomWriter getWriter(Object object) {
```

Comment:
create two methods
1) CustomWriter getOutfitWriter(Outfit outfit)
2) CustomWriter getSimpleWriter()

Object object - it is extremely bad input param

Reply:
fix